habitat_baselines/common/rollout_storage.py

```diff
@@ -4,237 +4,196 @@
+        self.buffers["observations"] = TensorDict()
 
         for sensor in observation_space.spaces:
-            self.observations[sensor] = torch.zeros(
-                num_steps + 1,
-                num_envs,
-                *observation_space.spaces[sensor].shape
+            self.buffers["observations"][sensor] = torch.from_numpy(
+                np.zeros(
+                    (
+                        numsteps + 1,
+                        num_envs,
+                        *observation_space.spaces[sensor].shape,
+                    ),
+                    dtype=observation_space.spaces[sensor].dtype,
+                )
             )
 
-        self.recurrent_hidden_states = torch.zeros(
-            num_steps + 1,
-            num_recurrent_layers,
+        self.buffers["recurrent_hidden_states"] = torch.zeros(
+            numsteps + 1,
             num_envs,
+            num_recurrent_layers,
             recurrent_hidden_state_size,
         )
 
-        self.rewards = torch.zeros(num_steps, num_envs, 1)
-        self.value_preds = torch.zeros(num_steps + 1, num_envs, 1)
-        self.returns = torch.zeros(num_steps + 1, num_envs, 1)
+        self.buffers["rewards"] = torch.zeros(numsteps + 1, num_envs, 1)
+        self.buffers["value_preds"] = torch.zeros(numsteps + 1, num_envs, 1)
+        self.buffers["returns"] = torch.zeros(numsteps + 1, num_envs, 1)
 
-        self.action_log_probs = torch.zeros(num_steps, num_envs, 1)
+        self.buffers["action_log_probs"] = torch.zeros(
+            numsteps + 1, num_envs, 1
+        )
         if action_space.__class__.__name__ == "ActionSpace":
             action_shape = 1
         else:
             action_shape = action_space.shape[0]
 
-        self.actions = torch.zeros(num_steps, num_envs, action_shape)
-        self.prev_actions = torch.zeros(num_steps + 1, num_envs, action_shape)
+        self.buffers["actions"] = torch.zeros(
+            numsteps + 1, num_envs, action_shape
+        )
+        self.buffers["prev_actions"] = torch.zeros(
+            numsteps + 1, num_envs, action_shape
+        )
         if action_space.__class__.__name__ == "ActionSpace":
-            self.actions = self.actions.long()
-            self.prev_actions = self.prev_actions.long()
+            self.buffers["actions"] = self.buffers["actions"].long()
+            self.buffers["prev_actions"] = self.buffers["prev_actions"].long()
+
+        self.buffers["masks"] = torch.zeros(
+            numsteps + 1, num_envs, 1, dtype=torch.bool
+        )
+
+        self.is_double_buffered = is_double_buffered
+        self._nbuffers = 2 if is_double_buffered else 1
+        self._num_envs = num_envs
+
+        assert (self._num_envs % self._nbuffers) == 0
 
-        self.masks = torch.zeros(num_steps + 1, num_envs, 1)
+        self.numsteps = numsteps
+        self.steps = [0 for _ in range(self._nbuffers)]
 
-        self.num_steps = num_steps
-        self.step = 0
+    @property
+    def step(self):
```

Comment:
There can be some confusion between methods `step` we are using for Env/Sim and property here. Maybe, we can use this moment to rename it to `step_id` or etc.